src/AutoRest.CSharp/Mgmt/Decorator/CodeModelExtension.cs

```diff
@@ -72,10 +72,38 @@ public static void VerifyAndUpdateFrameworkTypes(this IEnumerable<Schema> allSch
                         property.Schema.Type = AllSchemaTypes.Uri;
                     if (property.Language.Default.Name.SplitByCamelCase().Last().Equals("Duration") && property.Schema.Type == AllSchemaTypes.String)
                         throw new InvalidOperationException($"The {property.Language.Default.Name} property of {objSchema.Name} ends with \"Duration\" but does not use the duration format to be generated as TimeSpan type. Add \"format\": \"duration\" with directive in autorest.md for the property if it's ISO 8601 format like P1DT2H59M59S. Add both \"format\": \"duration\" and \"x-ms-format\": \"duration-constant\" if it's the constant format like 1.2:59:59.5000000. If the property does not conform to a TimeSpan format, please use \"x-ms-client-name\" to rename the property for the client.");
+                    // With UpdateAcronyms processing, the case of the name may be changed, so use ignore case comparison.
+                    // Do not use property.SerializedName=="type" so that we can still use x-ms-client-name to override the auto-renaming here if there is some edge case.
+                    if (property.Language.Default.Name.Equals("type", StringComparison.OrdinalIgnoreCase))
+                    {
+                        if (property.Schema.Type == AllSchemaTypes.String
+                            && objSchema.ContainsStringProperty("id")
+                            && objSchema.ContainsStringProperty("name"))
+                        {
+                            property.Language.Default.Name = "resourceType";
+                        }
```

Comment:
Here we are checking if a schema is a schema of a resource, we should have similar things in `Decorator/ResourceDetection`